freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+  suppose the `conditions` is `"B > 5"` then the `WHERE` clause will be equal to `WHERE A IS NOT NULL AND B > 5`.
+- If the client provided `limit` or `offset` parameter,  replace `<limit>` and `<offset>` with the given value respectively.
+- If the client provided `order_by` parameter, replace the `<ordering-1>, <ordering-2>, ..., <ordering-n>` with the
+  value given by the client, e.g. suppose the `order_by` parameter is `A ASC, B DESC` then the `ORDER BY` clause will
+  be equal to `ORDER BY A ASC, B DESC`.
+
+On how to use GViz API can refer to the appendix below.
+
+#### Update
+
+Update selected rows with the given value.
+
+The operation method should expect these parameters:
+| Parameter Name | Required | Remarks                                                                                                                                                                |
+| -------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `value`        | Yes      | the value that we're going to replace the rows with.                                                                                                                   |
+| `conditions`   | No       | Condition that must be true in order for the row to be updated in [Google Sheet Query Language][QueryLanguage] format. Defaults to update all columns if not provided. |
+
+To get the list of affected rows, we need to call GViz API with the following query:
+`SELECT A WHERE A IS NOT NULL [AND <condition>]`
+
+Note: `A` column is referring to the `_rid` column (which contains the row's index).
+
+After we get the list of indices that we need to update, call [spreadsheets.values.batchUpdate][BatchUpdateAPI] to
+update all the affected rows with the given `value`.
+
+#### Delete
+
+Delete rows that matched the given conditions.
+
+The operation method should expect these parameters:
+
+| Parameter Name | Required | Remarks                                                                                                                                                                |
+| -------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `conditions`   | No       | Condition that must be true in order for the row to be deleted in [Google Sheet Query Language][QueryLanguage] format. Defaults to delete all columns if not provided. |
+
+
+To get the list of affected rows, we need to call GViz API with the following query:
+`SELECT A WHERE A IS NOT NULL [AND <conditions>]`
+
+Note: `A` column is referring to the `_rid` column (which contains the row's index).
+
+After we get the list of row indices that we need to delete, call [spreadsheets.values.clear][ClearAPI] API to remove
+all the affected rows.
```

Comment:
What are the values to pass into the `Clear` API though? I think this part is missing how to use the `_rid` values into the clear API here.

Reply:
Updated to make it more obvious.